pagopa/io-app
# fix(IT Wallet): [SIW-1806] Fix state inconsistencies in wallet home screen

## Short description
This PR fixes an incorrect handling of states in the Wallet home screen

## List of changes proposed in this pull request
- Added `itwShouldRenderWalletReadyBannerSelector` selector
- Refactored `WalletCardsContainer` component to handle all the possible states of the wallet screen
- Added `selectWalletCategories` selector, which returns the card categories present in the wallet
- Added (a lot of) missing tests
- Added `walletHideCards` and `walletRestoreCards` actions to handle the hide/unhide of wallet cards, used for payment methods optimistic updates (cc: @Hantex9 @LeleDallas)

## How to test
Static checks should pass.
To test this PR you need to verify the correct behavior of the Wallet home screen in different scenarios. 
Check the table below to see the expected result in each scenario:

| Methods/bonuses | ITW | Preview |
| --- | --- | --- |
| ❌ No methods/bonuses | ❌ ITW disabled | <img src="https://github.com/user-attachments/assets/26c0f995-63d8-42c9-902b-a5a49ab79763" width="200" /> |
| ❌ No methods/bonuses | ✅ ITW enabled | <img src="https://github.com/user-attachments/assets/c80544d9-cb45-490c-a7cf-6de9879ba49e" width="200" /> |
| ❌ No methods/bonuses | ✅ ITW with credentials | <img src="https://github.com/user-attachments/assets/f02df9d4-d2ea-4fce-880c-a7ce5c24af83" width="200" /> |
| ✅ With methods/bonuses | ❌ ITW disabled |  <img src="https://github.com/user-attachments/assets/ac8784bb-9287-4925-9ca9-60699c4d89b8" width="200" /> |
| ✅ With methods/bonuses | ✅ ITW enabled | <img src="https://github.com/user-attachments/assets/63488a2c-7c06-4de4-a5c7-7db1f1ab8240" width="200" /> |
| ✅ With methods/bonuses | ✅ ITW with credentials | <img src="https://github.com/user-attachments/assets/7459a257-5703-4209-85f5-2facaef21695" width="200" /> |
